x-pack/plugins/ingest_manager/server/services/setup.ts

```diff
@@ -118,12 +119,23 @@ async function addPackageToConfig(
   config: AgentConfig,
   defaultOutput: Output
 ) {
+  const pkgkey = `${packageToInstall.name}/${packageToInstall.version}`;
   const packageInfo = await getPackageInfo({
     savedObjectsClient: soClient,
-    pkgkey: `${packageToInstall.name}-${packageToInstall.version}`,
+    pkgkey,
   });
-  await datasourceService.create(
-    soClient,
-    packageToConfigDatasource(packageInfo, config.id, defaultOutput.id, undefined, config.namespace)
+
+  const newDatasource = packageToConfigDatasource(
+    packageInfo,
+    config.id,
+    defaultOutput.id,
+    undefined,
```

Comment:
Can we update this signature to an object? Five parameters is a lot. Especially if some are optional.

Reply:
I did not modify this function here, just some prettier, and if you are good with I found this PR already containing a lot of change and keep it for another PR.

Reply:
👍 it's not a priority.